src/commonMain/kotlin/io/github/detekt/sarif4k/SarifDataBindings.kt

```diff
@@ -339,13 +341,41 @@ data class Address (
  *
  * Key/value pairs that provide additional information about the version control details.
  */
-@Serializable
-data class PropertyBag (
+@Serializable(with = PropertyBag.Companion::class)
+@JvmInline
+value class PropertyBag(
+    private val value: Map<String, Any?>,
+) : Map<String, Any?> by value {
     /**
      * A set of distinct strings that provide additional information.
      */
-    val tags: List<String>? = null
-)
+    val tags: List<String>?
+        get() = try {
+            val tagList = value["tags"] as Collection<*>?
+            tagList?.mapIndexed { position, it ->
+                try {
+                    it as String
+                } catch (e: ClassCastException) {
+                    throw IllegalStateException("the tag \"$it\" at the position $position is not a String", e)
+                }
+            }
+        } catch (e: ClassCastException) {
+            throw IllegalStateException("tags should be a collection", e)
```

Comment:
```suggestion
            throw IllegalStateException("Expected a Collection for the value of tags property: ${value["tags"]}.", e)
```
by toString [here](https://github.com/detekt/sarif4k/pull/101#discussion_r1499995588) I meant this ^.